internal/characters/thoma/skill.go

```diff
@@ -42,8 +45,10 @@ func (c *char) Skill(p map[string]int) action.ActionInfo {
 	}
 	c.Core.QueueParticle("thoma", count, attributes.Pyro, skillHitmark+c.Core.Flags.ParticleDelay)
 
-	shieldamt := (shieldpp[c.TalentLvlSkill()]*c.MaxHP() + shieldflat[c.TalentLvlSkill()])
-	c.genShield("Thoma Skill", shieldamt)
+	c.QueueCharTask(func() {
```

Comment:
technically you don't need char queue for this because he can't experience hitlag before this gets executed anyway